### PR TITLE
Update PXE building chapter

### LINKED ISSUE #323: Doc Redesign: Improve Task-Based Articles
## Problem description
Marcus did already a great job in restructuring the doc (see the `doc_redesign` branch). 👍 To go even further, we discussed some topics were we see improvements:

* Inconsistent structure of some articles
* Does not distinguish between concept-based and task-based

This issue is a summary of what we've discussed.

## Solution
As the structure is almost done, we came up with how we can improve "Building Images" part.

We distinguish between:

* **High level topics**
   these are topics like "Building a Docker Container Image". Usually, such articles describe how to build an image in format XYZ.
* **Low level topics**
  like "Using a Live Image on a Stick". These are the "building blocks" and are more or less independent from a specific image.

Each of these topics contains elements which should be consistent with a similar topics. Such a topic contains the following structural elements:

* **Title**
  The title starts with a verb ("Building/Creating...") to signal a topic which is task oriented.
* **Abstract**
  Leads in the topic. An itemized list contains the issues which are discussed.
* **The task**
  Contains the procedure with a step-by-step instruction.
* **Testing ...**
  Lists actions how the user can check the resulting image.
* **Additional Information**
  Optional links to related information that the user should be aware of.

**Update** The "Task" and "Testing" parts can be either steps or sections, depending on the amount of information.

### PR DESCRIPTION
**Pull request by schaefi**
Redesign PXE building chapter to follow style guide.
This Fixes #323

### CODE REVIEW

**Comment by schaefi** [comment]
@tomschr this one is not so easy. As you can see I described several different configuration flavors. Do you think we should put those into "working with images" chapters ? and if so how to deal with it in a good way ?

I used a section to describe all possible configuration parameters in order to provide an overview. next I created subsections describing use cases. In the end all use cases covered all possible configuration parameters. I'm not sure how a good split could look like ?

Feel free to commit to the branch if you'd like to demonstrate your idea

Thanks much

File: doc/source/building/build_pxe_root_filesystem.rst [hunk 1]
```diff
@@ -6,74 +6,103 @@ Build a PXE root File System Image
 
    build_pxe_root_filesystem/setup_pxe_bootserver
 
-.. hint::
```
Comment (tomschr) [inline doc/source/building/build_pxe_root_filesystem.rst]: Better use "Root" instead of "root" in the title "Build a PXE root File System Image"

**Comment by tomschr** [comment]
@schaefi I took your recommendation of "feel free to commit" to demonstrate my idea. 😉

I'm not 100% sure if this is the best solution, but let's see how you like it.

I did:

* Moved the `setup_pxe_bootserver.rst` to `source/building/working_with_images/` and added a references (in that file and in the referenced file)
* Move the "PXE Client Setup Configuration" section into a separate file `building/working_with_images/pxe_client_configuration.rst` and add references.
* Include the two files into 'working_with_images.rst`
* Fix some markup issues

The previous commits changes these. I didn't get any errors, maybe we should adjust some texts here and there.

Let me know what you think about this change.

**Comment by schaefi** [comment]
great thanks much, I like it. I squashed the commits and update the base commit message

**Pull request merged**